packages/cli/src/services/orchestration/main/MultiMainSetup.ee.ts

```diff
@@ -63,41 +63,43 @@ export class MultiMainSetup extends SingleMainSetup {
 	}
 
 	private async checkLeader() {
-		if (!this.redisPublisher.redisClient) return;
-
 		const leaderId = await this.redisPublisher.get(this.leaderKey);
 
-		if (!leaderId) {
-			this.logger.debug('Leadership vacant, attempting to become leader...');
-			await this.tryBecomeLeader();
+		if (leaderId === this.id) {
+			this.logger.debug(`[Instance ID ${this.id}] Leader is this instance`);
+
+			config.set('multiMainSetup.instanceType', 'leader');
```

Comment:
Maybe we need to check if not already the leader, we should fire the event stating that this instance became the leader; isn't this necessary?

Have you tried killing the current leader to see what happens? Was the transition ok?

Reply:
> Maybe we need to check if not already the leader, we should fire the event stating that this instance became the leader; isn't this necessary?

`leaderId === this.id` confirms that the instance _remained_ the leader, not that it _became_ the leader, so here there was no leader transition, so we do not emit a leadership change event. Unless you mean something else?

> Have you tried killing the current leader to see what happens? Was the transition ok?

I [manually tested](https://share.cleanshot.com/LLX90BFD) transitions. What is harder to test locally is spontaneous transitions.

Reply:
I reviewed the code and it seems ok but this call to `config.set('multiMainSetup.instanceType', 'leader');` seems redundant.